assembler_tools/hec-assembler-tools/README.md

```diff
@@ -0,0 +1,139 @@
+# HERACLES Code Generation Framework (Reference Implementation)
+
+The tools in this directory are the reference implementation of and Assembler codegenerator that takes a pre-generated Polynomial Instruction Set Architecture (P-ISA) program containing instructions that use an abstract, flat memory model for polynomial operations, such as those applied in homomorphic encryption (HE), and maps them to a corresponding set of instructions compatible with the HERACLES architecture, accounting for hardware restrictions, including memory management for the HERACLES memory model.
+
+## Table of Contents
+1. [Dependencies](#dependencies)
+2. [Inputs](#as_inputs)
+3. [Outputs](#as_outputs)
+   1. [Assembler Instruction Specs](#asm_specs)
+4. [Executing the Assembler](#executing_asm)
+   1. [Running for a Pre-Generated Kernel](#executing_single)
+   2. [Running for a Batch of Operations](#executing_batch)
+5. [Debug Tools](./debug_tools/README.md)
+
+## Dependencies <a name="dependencies"></a>
+
+This project is Python based. Dependencies for the project are as follows:
+
+- Python 3.10 or newer (Tested with 3.10)
+- Python pip 22.0 or newer
+- Requirements for the Python environment listed in file `requirements.txt`.
+
+  Install with `pip`:
+
+  ```bash
+  pip3 install -r requirements.txt
+  ```
+
+It is recommended to install Python dependencies and run inside a [Python virtual environment](https://virtualenv.pypa.io/en/stable/index.html) to avoid polluting the global Python.
+
+## Inputs <a name="as_inputs"></a>
+
+The assembler framework requires two inputs:
+
+- **Abstract P-ISA kernel**: a kernel of instructions in abstract P-ISA where the memory model is abstracted with a flat, infinite structure.
+
+- **Memory mapping metadata**: metadata information indicating the location where variable identifiers from the kernel are stored in the HERACLES HBM.
+
+Kernels and metadata are structured in comma-separated value (csv) files.
+
+P-ISA kernels, along with corresponding memory metadata required as input to the assembler, are generated by Python script `HERACLES-SEAL-isa-mapping/kernels/run_he_op.py` in the repo [HERACLES-SEAL-isa-mapping](https://github.com/IntelLabs/HERACLES-SEAL-isa-mapping)
```

Comment:
Reference to run_he_op.py

Reply:
I've updated README file. Check https://github.com/IntelLabs/encrypted-computing-sdk/commit/1d9652d1ba766047bfd4c949652efd98104f6a40